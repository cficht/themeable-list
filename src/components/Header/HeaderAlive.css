body {
  background-color: black;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
}

.Header {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 10px;
  margin-top: 20px;
  border: 10px solid rgb(137, 191, 209);
  text-align: center;
  background-color: lightblue;

  h1 {
    margin-block-start: .3em;
    margin-block-end: .3em;
  }

  h3 {
    margin-block-start: .5em;
    margin-block-end: 1em;
    font-style: italic;
  }
}