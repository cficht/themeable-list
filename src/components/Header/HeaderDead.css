.Header {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 10px;
  margin-top: 20px;
  border: 10px solid rgb(184, 42, 42);
  text-align: center;
  color: white;
  background-color: red;

  h1 {
    margin-block-start: .3em;
    margin-block-end: .3em;
  }

  h3 {
    margin-block-start: .5em;
    margin-block-end: 1em;
    font-style: italic;
  }
}