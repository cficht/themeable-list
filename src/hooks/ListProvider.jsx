import React, { createContext, useState, useEffect, useContext } from 'react';
import PropTypes from 'prop-types';

const ListContext = createContext();

export const ListProvider = ({ children }) => {
  const [characters, setCharacters] = useState([]);

  const chararcterTest = [
    {
      id: 1,
      name: 'Rick Sanchez',
      status: 'Alive',
      species: 'Human',
      image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg'
    },
    {
      id: 2,
      name: 'Morty Smith',
      status: 'Alive',
      species: 'Human',
      image: 'https://rickandmortyapi.com/api/character/avatar/2.jpeg'
    },
    {
      id: 3,
      name: 'Summer Smith',
      status: 'Alive',
      species: 'Human',
      image: 'https://rickandmortyapi.com/api/character/avatar/3.jpeg'
    }
  ];

  useEffect(() => {
    setCharacters(chararcterTest);
  }, []);

  return (
    <ListContext.Provider value={{ characters }}>
      {children}
    </ListContext.Provider>
  );
};

ListProvider.propTypes = {
  children: PropTypes.node
};

export const useCharacters = () => {
  const { characters } = useContext(ListContext);
  return characters;
};
