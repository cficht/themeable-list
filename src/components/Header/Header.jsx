import React from 'react';
import Toggle from '../Toggle/Toggle';
import { useStatus } from '../../hooks/ListProvider';
import stylesAlive from './HeaderAlive.css';
import stylesDead from './HeaderDead.css';

const Header = () => {
  const status = useStatus();
  const styles = status === 'alive' ? stylesAlive : stylesDead;

  return (
    <header className={styles.Header}>
      <h1>Rick and Morty</h1>
      <Toggle />
    </header>
  );
};

export default Header;
