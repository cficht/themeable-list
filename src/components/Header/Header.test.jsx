import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';
import { ListProvider } from '../../hooks/ListProvider';

describe('Header component', () => {
  it('renders Header', () => {
    const wrapper = shallow(
      <ListProvider>
        <Header />
      </ListProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
