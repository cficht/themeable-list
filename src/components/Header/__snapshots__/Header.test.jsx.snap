// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders Header 1`] = `
<ContextProvider
  value={
    Object {
      "characters": Array [],
      "handlePage": [Function],
      "page": 1,
      "status": "alive",
      "toggle": [Function],
    }
  }
>
  <Header />
</ContextProvider>
`;
